backend/app/adapter/db/user_changelog.go

```diff
@@ -29,28 +29,51 @@ WHERE "%s"=$1;`,
 	row := u.db.QueryRow(statement, user.Email)
 	lastViewedAt := time.Time{}
 	err := row.Scan(&lastViewedAt)
-	if err == sql.ErrNoRows {
-		return lastViewedAt, nil
-	} else if err != nil {
+	if err != nil {
 		return lastViewedAt, err
 	}
 
 	return lastViewedAt, nil
 }
 
 func (u UserChangeLogSQL) UpdateLastViewedAt(user entity.User, currentTime time.Time) (time.Time, error) {
+	lastViewedAt, err := u.GetLastViewedAt(user)
+	if err != nil {
+		return lastViewedAt, err
+	}
+
```

Comment:
I thought this is automatic and is not needed. Please let me know if I am wrong.